src/txn/txn_ckpt.c

```diff
@@ -1642,6 +1645,10 @@ __wt_txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[], bool waiting)
           __wt_panic(session, ret, "checkpoint can not fail when flush_tier is enabled"));
     WT_ERR(ret);
 
+    /* Trigger the checkpoint cleanup thread to remove the obsolete pages. */
+    if (checkpoint_cleanup)
```

Comment:
Then why this need to be triggered by a checkpoint? Checkpoint is running every one minute. Can we reduce our frequency of doing this?

Reply:
If we are in close, will the checkpoint cleanup be included in the last checkpoint?

Reply:
That option is purely used for testing to control the checkpoint cleanup operation in the tests to check whether it is really working to remove the obsolete pages or not. It is not invoked by default at the end of the checkpoint.

Currently, checkpoint cleanup is also scheduled every 60 seconds. It can be changed.

Reply:
Can we make it an debug option then? I don't want it to be accidentally activated in production.

Reply:
If this can run outside checkpoint, are we sure it is safe to run in parallel with checkpoint? It should consider checkpoint in the oldest check I believe.

Reply:
It has no issues running in parallel with the checkpoint. It is a checkpoint option, I can change add description to mention that it is used only for the testing scenarios.